### PR TITLE
chore(src): upgrade to openresty 1.21.4.2

### PR DESCRIPTION
**Pull request by chronolaw**
Summary

We need the related patches of OpenResty 1.21.4.2 to pass the ci tests.

Related PR : https://github.com/Kong/kong/pull/11360

Change logs:
- remove apple m1 compatible code (1.21.4.2 has already integrated it)

### CODE REVIEW

**Comment by chronolaw** [comment]
Reopen When OpenResty 1.21.4.2 is officially released.

File: t/001-tls.t [hunk 1]
```diff
@@ -415,8 +415,8 @@ qpvBFOY29Mlww88oWNGTA8QeW84Y1EJbRkHavzSsMFB73sxidQW0cHNC5t9RCKAQ
 uibeZgK1Yk7YQKXdvbZvXwrgTcAjCdbppw2L6e0Uy+OGgNjnIps8K460SdaIiA==
 -----END CERTIFICATE-----
 
---- error_log
-[lua] ssl_certificate_by_lua:2: ssl cert by lua is running!
+--- error_log eval
+qr/ssl_certificate_by_lua.*ssl cert by lua is running!/
```
Comment (samugi) [inline t/001-tls.t:419]: the log was: `[lua] ssl_certificate_by_lua(nginx.conf:30):2: ssl cert by lua is running!`

Comment (chronolaw) [reply t/001-tls.t:419]: What causes this change? any clue?

Comment (samugi) [reply t/001-tls.t:419]: I am trying to figure it out but not sure yet

Comment (samugi) [reply t/001-tls.t:419]: @chronolaw I think I found it: https://github.com/openresty/lua-nginx-module/pull/2008, see how the tests changed: https://github.com/openresty/lua-nginx-module/pull/2008/files#diff-9f25aa4f9bc139d6b6d65bf14649c6524911dc99ca7fc92e3337867a969bee1dR120

**Review by dndx — APPROVED** [review]

**Pull request merged**